tests/test_ml_trees.py

```
@@ -0,0 +1,91 @@
+    def test_entropy_criterion(self):
+        x, t = linearly_separable_classification(seed=2)
+
+        tree = ClassificationTree(max_depth=6, criterion="entropy", seed=1)
+        tree.fit(x, t)
+        predict = tree.predict(x)
+
+        accuracy = float((predict.data == t).mean())
+        self.assertGreater(accuracy, 0.9)
+
+    def test_invalid_criterion_rejected(self):
+        with self.assertRaises(Exception):
```

Ruff (0.15.15)

[warning] Do not assert blind exception: `Exception`

(B017)